lona/html/data_binding/select.py

```diff
@@ -80,73 +154,109 @@ def readonly(self, new_value):
                 if node.tag_name != 'option':
                     continue
 
-                values.append(
-                    (node.attributes.get('value', ''),
-                     node.get_text(),
-                     'selected' in node.attributes),
-                )
+                options += (node, )
 
-            return values
+            return options
 
-    @values.setter
-    def values(self, new_values):
+    @options.setter
+    def options(self, new_options):
         with self.lock:
-            self.clear()
-
-            for i in new_values:
-                value, name, selected = (list(i) + [False])[0:3]
-                option_node = Option(str(name), value=str(value))
+            self.nodes.clear()
 
-                if selected:
-                    option_node.attributes['selected'] = ''
+            if not isinstance(new_options, (list, tuple)):
+                new_options = [new_options]
 
-                self.append(option_node)
+            for option in new_options:
+                self.add_option(option)
 
-    # value
+    # selected options
     @property
-    def value(self):
+    def selected_options(self):
         with self.lock:
-            value = []
+            options = ()
+
+            for option in self.options:
+                if not option.selected:
+                    continue
 
-            for option in self.nodes:
-                if 'selected' in option.attributes:
-                    value.append(option.attributes['value'])
+                options += (option, )
 
-            if not self.multiple and not value and self.nodes:
-                option = self.nodes[0]
+            if not options and self.options and not self.multiple:
+                return (self.options[0], )
 
-                value.append(option.attributes['value'])
+            return options
 
-            if not value:
-                if not self.multiple:
-                    return None
+    # value
+    @property
+    def value(self):
+        with self.lock:
+            values = ()
 
-                return value
+            for option in self.selected_options:
+                values += (option.value, )
 
             if not self.multiple:
-                value = value.pop()
+                if not values:
+                    return self.options[0].value
 
-            return value
+                return values[-1]
+
+            return values
 
     @value.setter
     def value(self, new_value):
-        if not isinstance(new_value, list):
-            new_value = [new_value]
+        with self.lock:
+            if not isinstance(new_value, list):
+                new_value = [new_value]
+
+            old_values = self.values
+
+            for value in new_value:
+                if value not in old_values:
+                    raise RuntimeError(f'unknown value: {value}')
 
+            for option in self.options:
+                option._set_selected(option.value in new_value)
```

Comment:
What if there are several options with the same value? I think we should select only the first one (as in browser).

Reply:
Yes i agree. Like in your test